packages/ui/src/machines/authenticator/actors/signIn.ts

```diff
@@ -104,6 +104,11 @@ export function signInActor({ services }: SignInMachineOptions) {
                     actions: ['setUser', 'setChallengeName'],
                     target: '#signInActor.setupTOTP',
                   },
+                  {
+                    cond: 'shouldSelectMfaType',
+                    actions: ['setUser', 'setChallengeName'],
```

Comment:
We shouldn't `setUser` here (or arguably even `challengeName`), since we're intentionally not progressing further if we come across `SELECT_MFA_TYPE`.